File: Blueprints/BlueprintDefinitions/vs2026/SimpleApplicationLoadBalancer/template/test/BlueprintBaseName.1.Tests/FunctionTest.cs
using Xunit;
using Amazon.Lambda.Core;
using Amazon.Lambda.ApplicationLoadBalancerEvents;
using Amazon.Lambda.TestUtilities;

namespace BlueprintBaseName._1.Tests;

public class FunctionTest
{
    [Fact]
    public void TestSampleFunction()
    {
        var function = new Function();
        var context = new TestLambdaContext();
        var request = new ApplicationLoadBalancerRequest();
        var response = function.FunctionHandler(request, context);

        Assert.Equal(200, response.StatusCode);
        Assert.Contains("Hello World from Lambda", response.Body);
    }
}

File: Blueprints/BlueprintDefinitions/vs2026/NativeAOTServerless-FSharp/template/test/BlueprintBaseName.1.Tests/FunctionTest.fs
namespace BlueprintBaseName._1.Tests


open Xunit
open Amazon.Lambda.TestUtilities
open Amazon.Lambda.APIGatewayEvents
open BlueprintBaseName._1.Function


module FunctionTest =

    [<Fact>]
    let ``Call HTTP GET on Root``() =
        let context = TestLambdaContext()
        let request = APIGatewayProxyRequest()
        let response = GetFunctionHandler request context

        Assert.Equal(200, response.StatusCode)
        Assert.Equal("Hello AWS Serverless", response.Body)

